common/sgx/collateral.c

```diff
@@ -325,14 +325,19 @@ oe_result_t oe_validate_revocation_list(
                 "Failed to read CRL. %s",
                 oe_result_str(result));
         }
-        // Otherwise, CRL should have v3 structure which is Der encoded
+        // Otherwise, CRL should have v3 structure which is DER encoded
         else
         {
+            size_t der_data_size = sgx_endorsements->items[i].size;
+
+            // If DER CRL buffer has null terminator, need to remove it before
```

Comment:
In the previous code. (sgx_endorsements->items[i].size == 0) error case is covered `oe_crl_read_der()`. The new code does not cover the case and will reference to data[der_data_size-1] without checking.

Please check if der_data_size >= 1 before proceed.

Reply:
Nice catch!